.github/workflows/ubuntu-ci-x86_64-gnu.yaml

```diff
@@ -0,0 +1,192 @@
+name: ubuntu-ci-c6a-x86_64-gnu-build
```

Comment:
This is almost a copy of the `.github/workflows/ubuntu-ci-x86_64-intel.yaml` workflow. There are subtle differences and I thought it might be better to keep them apart, even though this means some amount of code duplication. We could move some of the steps into a common/shared workflow file and draw from there in the future ...